.github/workflows/update-project-field.yml

```diff
@@ -0,0 +1,71 @@
+name: Update Project Field to Match Label
+
+# Triggers the workflow when a label is added to an issue.
+on:
+  issues:
+    types:
+      - labeled
+
+# Details needed to update a project field.
+env:
+  ORGANIZATION: "Automattic"
+  PROJECT_NUMBER: 391
+  PRIORITY_PREFIX: "[Pri]"
+  PRIORITY_FIELD: "Priority"
+
+jobs:
+
+  # Updates project fields to match label changes.
+  update_project_field: 
+    runs-on: ubuntu-latest
+    steps:
+
+      # Checks if the issue is already in the project. If not, the next steps will be skipped.
+      - name: Check if issue is in project
```

Comment:
> I think our best bet is having the first step in this workflow be an API call that checks and evaluates if the current issue is on our project.

So after a lot of trial and error, I was able to cobble something together 😅 The GraphQL piece iterates through the issue's project, looks for a match on the project number, and stores a boolean value in a new variable `ISSUE_IN_PROJECT`, which we use later in the conditional statements.

Reply:
Nice! 🚀